VSGitDiff/GitDiff.cs

```diff
@@ -36,89 +41,51 @@ internal sealed class GitDiff
         public static readonly Guid CommandSet = new Guid("be3bf61c-3d52-4384-8f96-388ecbfe929e");
 
         /// <summary>
-        /// VS Package that provides this command, not null.
-        /// </summary>
-        private readonly Package package;
-
-        /// <summary>
-        /// Gets the instance of the command.
+        /// Initializes static services of the command.
+        /// Adds our command handlers for menu (commands must exist in the command table file).
         /// </summary>
-        public static GitDiff Instance
+        /// <param name="package">Owner package, not null.</param>
+        public static async System.Threading.Tasks.Task InitializeAsync(AsyncPackage package)
         {
-            get;
-            private set;
-        }
+            await package.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Choice to add the switch to the UI thread based on the following documents:

- [Migrate to AsyncPackage](https://github.com/Microsoft/VSSDK-Extensibility-Samples/tree/master/AsyncPackageMigration)
- [Asynchronous and multithreaded programming within VS using the JoinableTaskFactory](https://devblogs.microsoft.com/premier-developer/asynchronous-and-multithreaded-programming-within-vs-using-the-joinabletaskfactory/)
- [How to: Provide an asynchronous Visual Studio service](https://docs.microsoft.com/en-us/visualstudio/extensibility/how-to-provide-an-asynchronous-visual-studio-service?view=vs-2019)

In all the examples, the switch to the UI thread (calling `await package.JoinableTaskFactory.SwitchToMainThreadAsync()`) seems to be done before calls to `GetServiceAsync`.

For `GetGlobalService`, the second article seems to indicate that it definitely needs to be done here as well, so this one is included after the call as well:

> It is not always obvious whether some VS API you’re calling may require marshaling to the UI thread. Here is a list of types, members, or tips to help identify when you should explicitly switch to the UI thread yourself using SwitchToMainThreadAsync() before calling into VS:
> 
> 1. Package.GetGlobalService()
> 2. Casting an object to any other type (object to IVsHierarchy for example). If the object being cast is a native COM object, the cast may incur a call to IUnknown.QueryInterface, and will most likely require the UI thread.
> 3. Almost any call to an IVs* interface.